Inputting/InputParserBuilder.cs

```diff
@@ -0,0 +1,368 @@
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// Adds a remapped analog stick. This is a shortcut for adding remapped analog inputs for
+        /// "up"/"down"/"left"/"right", plus configuring respective conflicts, like inputting opposing directions.
+        /// </summary>
+        /// <param name="up">analog input that maps to "up"</param>
+        /// <param name="down">analog input that maps to "down"</param>
+        /// <param name="left">analog input that maps to "left"</param>
+        /// <param name="right">analog input that maps to "right"</param>
+        /// <param name="spinl">(optional) analog input that maps to "spinl", or <c>null</c> if none.</param>
+        /// <param name="spinr">(optional) analog input that maps to "spinr", or <c>null</c> if none.</param>
+        /// <param name="mapsToPrefix">prefix for the analog stick that the remapping names map to,
+        /// which will get prepended to "up"/"down"/"left"/"right"</param>
+        public InputParserBuilder RemappedAnalogStick(
+            string up, string down, string left, string right,
+            string? spinl, string? spinr, string mapsToPrefix)
+        {
+            string targetUp = mapsToPrefix + "up";
+            string targetDown = mapsToPrefix + "down";
+            string targetLeft = mapsToPrefix + "left";
+            string targetRight = mapsToPrefix + "right";
+            RemappedAnalogInputs((up, targetUp), (down, targetDown), (left, targetLeft), (right, targetRight));
+            Conflicts((up, down), (left, right));
+            if (spinl != null)
+            {
+                RemappedButtons((spinl, mapsToPrefix + "spinl"));
+                Conflicts((spinl, up), (spinl, down), (spinl, left), (spinl, right));
+            }
+            if (spinr != null)
+            {
+                RemappedButtons((spinr, mapsToPrefix + "spinr"));
+                Conflicts((spinr, up), (spinr, down), (spinr, left), (spinr, right));
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// Add a D-pad. This is a shortcut for adding buttons for "up"/"down"/"left"/"right",
+        /// plus configuring respective conflicts, like inputting opposing directions.
+        /// </summary>
+        /// <param name="prefix">prefix for the D-pad, which will get prepended to "up"/"down"/"left"/"right"</param>
+        public InputParserBuilder DPad(string prefix)
+        {
+            string up = prefix + "up";
+            string down = prefix + "down";
+            string left = prefix + "left";
+            string right = prefix + "right";
+            Buttons(up, down, left, right);
+            Conflicts((up, down), (left, right));
+            return this;
+        }
+
+        /// <summary>
+        /// Add an aliased D-pad. This is a shortcut for adding aliased buttons for "up"/"down"/"left"/"right",
```

Comment:
This might come in handy one day but could be accomplished much more simply by adding an optional mapToPrefix to the above function

Reply:
not if you have custom prefixes like n/s/e/w